riley-research/IPSA
# Add firestore and firebase storage for uploading files

Added Firebase integration to manage data uploads and storage.

### What changed?

- Updated `package.json` and `package-lock.json` to add Firebase and related dependencies.
- Modified `App.tsx` to include a new `initializeGlobals` file.
- Enhanced `BulkDataUploadOptions.tsx` with improved file type labels and MobX observer integration.
- Updated `DataSelect.tsx` to handle file uploads and manage state using MobX.
- Created `firebase.ts` to initialize Firebase configuration.
- Introduced `initializeGlobals.ts` for global store initialization.
- Implemented `DataUploadStore.ts` with methods for uploading and listening to files.
- Enhanced `UserStore.ts` with user-specific metadata.
- Added new types in `DataUploadTypes.tsx` for better type safety.

### How to test?

1. Run `npm install` to install the new dependencies.
2. Start the app using `npm start`.
3. Verify that file uploads in the `BulkDataUploadOptions` component are managed correctly through Firebase.
4. Check that the uploaded files appear in the Firebase Firestore database.
5. Ensure that the files' state is managed and updated correctly in the app.
